be/src/io/shared_buffered_input_stream.cpp

```diff
@@ -194,22 +194,27 @@ Status SharedBufferedInputStream::set_io_ranges(const std::vector<IORange>& rang
     }
 }
 
-StatusOr<SharedBufferedInputStream::SharedBuffer*> SharedBufferedInputStream::find_shared_buffer(size_t offset,
-                                                                                                 size_t count) {
+StatusOr<SharedBufferedInputStream::SharedBufferPtr> SharedBufferedInputStream::find_shared_buffer(size_t offset,
+                                                                                                   size_t count) {
     auto iter = _map.upper_bound(offset);
     if (iter == _map.end()) {
         return Status::RuntimeError("failed to find shared buffer based on offset");
     }
-    SharedBuffer& sb = iter->second;
-    if ((sb.offset > offset) || (sb.offset + sb.size) < (offset + count)) {
+    SharedBufferPtr sb = iter->second;
+    if ((sb->offset > offset) || (sb->offset + sb->size) < (offset + count)) {
         return Status::RuntimeError("bad construction of shared buffer");
     }
-    return &sb;
+    return sb;
 }
 
-Status SharedBufferedInputStream::get_bytes(const uint8_t** buffer, size_t offset, size_t nbytes) {
-    ASSIGN_OR_RETURN(auto ret, find_shared_buffer(offset, nbytes));
-    SharedBuffer& sb = *ret;
+Status SharedBufferedInputStream::get_bytes(const uint8_t** buffer, size_t offset, size_t nbytes,
```

Comment:
> Can you add some comments about this function?
> 
> Looks like this function will fill `buffer` and return `SharedBufferPtr` at the same time.

I have add some comments. In fact, the `SharedBufferPtr` will not be returned because we only changed it as a local value, which will not affect the old value which passes to the function.

Reply:
> And when `shared_buffer` will be nullptr

We use non-null pointer of shared_buffer to avoid finding shared buffer repeatedly. So, until now no null shared_buffer passed to the function. And we check its nullable to adaptor some other cases in the future.